File: include/apl/errors/RankError.java
package APL.errors;

import APL.types.*;

public class RankError extends APLError {
  // public RankError(std::string s) {
  //   super(s);
  // }
  public RankError(std::string s, Tokenable fun) {
    super(s, fun);
  }
  public RankError(std::string s, Callable fun, Tokenable cause) {
    super(s, fun, cause);
  }
}

File: APL/errors/RankError.cpp
#include <APL/RankError.h>

namespace APL::errors
{
    // using namespace APL::types;

    RankError::RankError(std::string const& s, std::shared_ptr<APL::types::Tokenable> fun) : APLError(s, fun)
    {
    }

    RankError::RankError(std::string const& s, std::shared_ptr<APL::types::Callable> fun, std::shared_ptr<APL::types::Tokenable> cause) : APLError(s, fun, cause)
    {
    }
}
